File: BackEnd_Learn/Java/daily_book/src/main/java/com/dsm/daily_book/security/jwt/auth/AuthDetailsService.java
package com.dsm.daily_book.security.jwt.auth;

import com.dsm.daily_book.entity.user.UserRepository;
import com.dsm.daily_book.exception.user.UserNotFoundException;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AuthDetailsService implements UserDetailsService {
    private final UserRepository userRepository;

    @Override
    public AuthDetails loadUserByUsername(String userId) throws UsernameNotFoundException {
        return userRepository.findById(Integer.parseInt(userId))
                .map(AuthDetails::new)
                .orElseThrow(UserNotFoundException::new);
    }
}

File: BackEnd_Learn/kotlin/kopring/src/main/kotlin/io/github/leeseojune53/kopring/global/security/auth/AuthDetailsService.kt
package io.github.leeseojune53.kopring.global.security.auth

import io.github.leeseojune53.kopring.domain.user.domain.User
import io.github.leeseojune53.kopring.domain.user.domain.repositories.UserRepository
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.stereotype.Service

@Service
class AuthDetailsService(
    private val userRepository: UserRepository,
) : UserDetailsService {

    override fun loadUserByUsername(username: String?): UserDetails {
        val user: User = userRepository.findByName(username!!) ?: throw RuntimeException()
        return AuthDetails(user.name)
    }


}